### PR TITLE
Add headers and basic authentication

### PR DESCRIPTION
**Pull request by rsevilla87**
Provide basic authentication and a mechanism to add custom headers to the API requests.

Signed-off-by: Raul Sevilla <rsevilla@redhat.com>

### CODE REVIEW

**Comment by beorn7** [comment]
Sorry for the delay in reviewing this, I'm currently on vacations. Will have a look next week.

In the meantime, @lilic @bwplotka , perhaps you have an opinion on this?

File: api/client.go [hunk 1]
```diff
@@ -43,6 +43,15 @@ type Config struct {
 	// RoundTripper is used by the Client to drive HTTP requests. If not
 	// provided, DefaultRoundTripper will be used.
 	RoundTripper http.RoundTripper
```
Comment (lilic) [inline api/client.go:45]: The roundtripper is where you inject any authentication or any credentials you want to set for the requests.

Comment (rsevilla87) [reply api/client.go:45]: oh, you're right. I didn't realize about this one.
Thanks for the heads up!

Comment (lilic) [reply api/client.go:45]: No problem! :)

**Comment by beorn7** [comment]
Thanks for jumping in and resolving this @lilic .

**Pull request closed without merging**